internal/gatewayapi/runner/runner.go

```diff
@@ -336,6 +429,90 @@ func (r *Runner) subscribeAndTranslate(sub <-chan watchable.Snapshot[string, *re
 				}
 			}
 
+			// Store the stauses of any objects with multiple parents/ancestors (Routes or Policies)
+			// once after all parents are processed. This ensures the status update is atomic.
+			for _, httpRoute := range accumulatedResult.HTTPRoutes {
+				key := utils.NamespacedName(httpRoute)
+				r.ProviderResources.HTTPRouteStatuses.Store(key, &httpRoute.Status)
+				httpRouteStatusCount++
+				delete(keysToDelete.HTTPRouteStatus, key)
+				r.keyCache.HTTPRouteStatus[key] = true
+			}
+			for _, grpcRoute := range accumulatedResult.GRPCRoutes {
+				key := utils.NamespacedName(grpcRoute)
+				r.ProviderResources.GRPCRouteStatuses.Store(key, &grpcRoute.Status)
+				grpcRouteStatusCount++
+				delete(keysToDelete.GRPCRouteStatus, key)
+				r.keyCache.GRPCRouteStatus[key] = true
+
+			}
+			for _, tlsRoute := range accumulatedResult.TLSRoutes {
+				key := utils.NamespacedName(tlsRoute)
+				r.ProviderResources.TLSRouteStatuses.Store(key, &tlsRoute.Status)
+				tlsRouteStatusCount++
+				delete(keysToDelete.TLSRouteStatus, key)
+				r.keyCache.TLSRouteStatus[key] = true
+			}
+			for _, tcpRoute := range accumulatedResult.TCPRoutes {
+				key := utils.NamespacedName(tcpRoute)
+				r.ProviderResources.TCPRouteStatuses.Store(key, &tcpRoute.Status)
+				tcpRouteStatusCount++
+				delete(keysToDelete.TCPRouteStatus, key)
+				r.keyCache.TCPRouteStatus[key] = true
+			}
+			for _, udpRoute := range accumulatedResult.UDPRoutes {
+				key := utils.NamespacedName(udpRoute)
+				r.ProviderResources.UDPRouteStatuses.Store(key, &udpRoute.Status)
+				udpRouteStatusCount++
+				delete(keysToDelete.UDPRouteStatus, key)
+				r.keyCache.UDPRouteStatus[key] = true
+			}
+			for _, backendTLSPolicy := range accumulatedResult.BackendTLSPolicies {
+				key := utils.NamespacedName(backendTLSPolicy)
+				if len(backendTLSPolicy.Status.Ancestors) > 0 {
```

Comment:
Do we still need this branch when the status contains all the parents from all the GatewayClasses? cc @arkodg 

I think we should remove this otherwise the parent from EG controller can't be deleted after the parents of the policy changed.